src/key.js

```diff
@@ -771,6 +771,84 @@ Key.prototype.verifyAllUsers = async function(keys) {
   return results;
 };
 
+/**
+ * generate a new subkey to a Key
+ * @param  {Object} options   (optional) options for creating subkey, e.g. {sign: true, passphrase: '123', algorithm:..., curve:...}
+ *
+ * @returns {Promise<module:key.SubKey>} return the subkey if successful.
+ * @async
+ */
+Key.prototype.generateSubkey = async function(options){
```

Comment:
I don't think this function name makes it clear that the subkey is not only going to be generated, but also added to the key. So I would suggest combining this function with the one under it, renaming this to "addSubkey" and allowing an optional parameter of a subkey packet. If the packet is present, add the packet. Otherwise, generate it with the options.

Reply:
Yes, It's used to make me a headache for a while.. I think `generate` and `add` is different meaning. This is why using `openpgp.generateKey` instead of `openpgp.addKey`.

But the `Key.prototype.generateSubkey` instance method which means  "generate the subkey on the key", the subkey relates(belongs) to the key.

The `SubKey.generate` static method which means "generate an independent subkey" if it is exists.

Maybe I should add the `Subkey.generate` method and modify the constructor of Subkey:
generate a subkey if no packet parameter passed.